src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.darts.courthouse.service;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.common.entity.Courthouse;
+import uk.gov.hmcts.darts.courthouse.CourthouseRepository;
+import uk.gov.hmcts.darts.courthouse.mapper.CourthouseToCourthouseEntityMapper;
+
+import java.util.List;
+
+@AllArgsConstructor
+@Service
+@Slf4j
+public class CourthouseServiceImpl implements CourthouseService {
+
+    private CourthouseRepository repository;
+
+    private CourthouseToCourthouseEntityMapper mapper;
+
+    @Override
+    public void deleteCourthouseById(Integer id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public Courthouse amendCourthouseById(uk.gov.hmcts.darts.courthouse.model.Courthouse courthouse, Integer id) {
+        Courthouse originalEntity = repository.getReferenceById(id);
+
+        originalEntity.setCourthouseName(courthouse.getCourthouseName());
+        originalEntity.setCode(courthouse.getCode().shortValue());
+
+        return repository.saveAndFlush(originalEntity);
+    }
+
+    @Override
+    public Courthouse getCourtHouseById(Integer id) {
+        return repository.getReferenceById(id);
+    }
+
+    @Override
+    public List<Courthouse> getAllCourthouses() {
+        return repository.findAll();
+    }
+
+    @Override
+    public Courthouse addCourtHouse(uk.gov.hmcts.darts.courthouse.model.Courthouse courthouse) {
+        Courthouse mappedEntity = this.mapper.mapToEntity(courthouse);
```

Comment:
we would want to confirm that the name of the courthouse doesn't already exist, and throw an error if it does.

Reply:
Just an update Barry's schema change is only making code unique.

Reply:
@HMCTSDeveloper v20 proposed schema looks like

```
CREATE TABLE DARTS.moj_courthouse
(moj_crt_id                INTEGER					 NOT NULL
,courthouse_code           SMALLINT                                    UNIQUE
,courthouse_name           CHARACTER VARYING         NOT NULL          UNIQUE
);
```